### ISSUE TITLE
Structure

### ISSUE THREAD

**Issue by gruberjl** [gruberjl/projectManagement#6]
Projects - overarching project
Phase - parts of projects (discovery, implement, close). Might want them to be predefined to make task dependency easier.
Milestones - I may want predefined milestones and dynamic phases. A task could have a milestone as a dependency.
Processes - List of tasks. Are added to projects
Tasks - In a process. Can have multiple dependencies of tasks or milestones.
task resolution - A task may have failed or succeeded. or a task may be to get SOW signed and the client decided to not go forward.
issue - A problem that aroused during the project. Maybe in scope or out of scope. Could have a link to a task. Can mark task not completable until the issue is resolved. At the end of the project, there should be an issue review to determine if you need new tasks in any processes or anything like that.

**Comment by gruberjl** [comment]
Can you update a project based on discovery? Do you create a new project based on discovery?

Perform a discovery to create a SOW, if SOW is signed than continue with the project. If you find something during the discovery add it to the project.

Example: during O365 migration discovery I find a mailbox that is too large to migrate I need to add tasks to split the mailbox. Unfortunately, there is a large calculation based on the size of the mailbox.

I would need a task to return data, then I would need a workflow to process the data and create tasks or add processes or whatever.

I'm thinking of having 1 project, then updating the other phases based on discovered information.

Another workflow for task completion - when a SOW does not get approved the rest of the project is cancelled and a new process should be added (email the client a closing email, inform other project members) mark the project as cancelled with a template message (and allow for comments).

**Comment by gruberjl** [comment]
Meetings should be cooked in as well. Could you create a PowerPoint or a PDF that summarizes everything that has happened since the last meeting?

**Comment by gruberjl** [comment]
There may be a workflow of - after task is complete wait X time before doing next task.

For example when changing the primary smtp address for users:
Task 1: Send email to user's saying this change is happening
Task 2: depends on Task 1. Wait for a week or so after Task 1 is complete then make the change on task 2

**Comment by gruberjl** [comment]
Tasks should have man hours (amount of time to bill)
and actual hours (amount of time until it is complete)
That way a project can have an  estimated start / end time & estimate / quote the cost.

As an added plus I could have roles and each role could have a different cost, then when you can have 5 hours at $100 and 10 hours at $500.
